infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogCompaction.java

```diff
@@ -0,0 +1,54 @@
+package org.corfudb.infrastructure.log;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.util.MetricsUtils;
+
+import java.util.concurrent.*;
+
+/**
+ * Scheduled log compaction manager.
+ * Execute log compaction safely (prevent scheduled executor corruption if the task throws an exception)
+ * https://stackoverflow.com/questions/6894595/scheduledexecutorservice-exception-handling
+ */
+@Slf4j
+public class StreamLogCompaction {
+    public static final String STREAM_COMPACT_METRIC = StreamLogCompaction.class.getName();
+
+    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+            .setDaemon(true)
+            .setNameFormat("LogUnit-Maintenance-%d")
+            .build();
+
+    /**
+     * How many times log compaction executed
+     */
+    private final Counter gc = CorfuRuntime.getDefaultMetrics().counter(STREAM_COMPACT_METRIC);
```

Comment:
where is this used ? Could the name be more intuitive and not an abbreviation ?

Reply:
This counter is used inside this class, to count the number of compactions. is  `counter` ok? Welcome to any better names

Reply:
Can you rename it gcCounter?